File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteTeacherColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { InstituteTeacherRow } from "./InstituteTeacherRow";

export interface InstituteTeacherColumns {
    Id: Column<InstituteTeacherRow>;
    TeacherPrn: Column<InstituteTeacherRow>;
    SchoolName: Column<InstituteTeacherRow>;
    ClassTitle: Column<InstituteTeacherRow>;
    SubjectTitle: Column<InstituteTeacherRow>;
    Description: Column<InstituteTeacherRow>;
    AcademicYearName: Column<InstituteTeacherRow>;
    InsertDate: Column<InstituteTeacherRow>;
    InsertUserId: Column<InstituteTeacherRow>;
    UpdateDate: Column<InstituteTeacherRow>;
    UpdateUserId: Column<InstituteTeacherRow>;
    IsActive: Column<InstituteTeacherRow>;
}

export class InstituteTeacherColumns extends ColumnsBase<InstituteTeacherRow> {
    static readonly columnsKey = 'Institute.InstituteTeacher';
    static readonly Fields = fieldsProxy<InstituteTeacherColumns>();
}

File: GXpert/GXpert.Web/Modules/Institute/InstituteTeacher/InstituteTeacherColumns.cs
﻿using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Institute.Columns;

[ColumnsScript("Institute.InstituteTeacher")]
[BasedOnRow(typeof(InstituteTeacherRow), CheckNames = true)]
public class InstituteTeacherColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string TeacherPrn { get; set; }
    public string SchoolName { get; set; }
    public string ClassTitle { get; set; }
    public string SubjectTitle { get; set; }
    [EditLink]
    public string Description { get; set; }
    public string AcademicYearName { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public bool IsActive { get; set; }
}